<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>509</width>
    <height>464</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>10</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>SUDOKU SOLVING</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="n0_0">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>40</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n0_1">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>40</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n0_2">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>40</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n0_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>40</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n0_4">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>40</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n0_5">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>40</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n0_6">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>40</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n0_7">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>40</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n0_8">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>40</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n1_0">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>80</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n1_1">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>80</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n1_2">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>80</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n1_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>80</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n1_4">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>80</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n1_5">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>80</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n1_6">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>80</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n1_7">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>80</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n1_8">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>80</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n2_0">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>120</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n2_1">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>120</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n2_2">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>120</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n2_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>120</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n2_4">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>120</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n2_5">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>120</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n2_6">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>120</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n2_7">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>120</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n2_8">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>120</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n3_0">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>170</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n3_1">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>170</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n3_2">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>170</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n3_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>170</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n3_4">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>170</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n3_5">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>170</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n3_6">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>170</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n3_7">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>170</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n3_8">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>170</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n4_0">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>210</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n4_1">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>210</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n4_2">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>210</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n4_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>210</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n4_4">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>210</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n4_5">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>210</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n4_6">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>210</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n4_7">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>210</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n4_8">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>210</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n5_0">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>250</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n5_1">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>250</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n5_2">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>250</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n5_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>250</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n5_4">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>250</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n5_5">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>250</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n5_6">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>250</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n5_7">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>250</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n5_8">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>250</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n6_0">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>300</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n6_1">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>300</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n6_2">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>300</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n6_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>300</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n6_4">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>300</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n6_5">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>300</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n6_6">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>300</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n6_7">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>300</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n6_8">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>300</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n7_0">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>340</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n7_1">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>340</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n7_2">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>340</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n7_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>340</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n7_4">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>340</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n7_5">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>340</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n7_6">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>340</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n7_7">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>340</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n7_8">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>340</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n8_0">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>380</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n8_1">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>380</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n8_2">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>380</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n8_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>380</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n8_4">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>380</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n8_5">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>380</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n8_6">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>380</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n8_7">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>380</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="n8_8">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>380</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>150</y>
      <width>391</width>
      <height>21</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>280</y>
      <width>391</width>
      <height>21</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>410</y>
      <width>391</width>
      <height>21</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>391</width>
      <height>21</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_5">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>30</y>
      <width>20</width>
      <height>391</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_6">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>30</y>
      <width>20</width>
      <height>391</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_7">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>30</y>
      <width>20</width>
      <height>391</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_8">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>20</width>
      <height>391</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="reset">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>230</y>
      <width>71</width>
      <height>191</height>
     </rect>
    </property>
    <property name="text">
     <string>Đặt lại</string>
    </property>
   </widget>
   <widget class="QPushButton" name="solve">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>30</y>
      <width>71</width>
      <height>191</height>
     </rect>
    </property>
    <property name="text">
     <string>Giải</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>509</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
